pkg/controller/certificates/certificate_request.go

```diff
@@ -741,7 +741,15 @@ func (c *certificateRequestManager) buildCertificateRequest(log logr.Logger, crt
 			IssuerRef: crt.Spec.IssuerRef,
 			IsCA:      crt.Spec.IsCA,
 		},
-	}, nil
+	}
+
+	if apiutil.IssuerKind(crt.Spec.IssuerRef) == apiutil.IssuerSelfSigned {
```

Comment:
This will never be true - `kind` is `Issuer` or `ClusterIssuer` etc.

I also think we should *always* set this annotation - we can't know what implementations will require it.